include/appbase/default_executor.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <appbase/application_base.hpp>
+
+namespace appbase {
+
+   class default_executor {
+   public:
+      template <typename Func>
+      auto post( int priority, Func&& func ) {
+         return boost::asio::post(io_serv, pri_queue.wrap(priority, std::forward<Func>(func)));
+      }
+
+      /**
+       * Provide access to execution priority queue so it can be used to wrap functions for
+       * prioritized execution.
+       *
+       * Example:
+       *   boost::asio::steady_timer timer( app().get_io_service() );
+       *   timer.async_wait( app().get_priority_queue().wrap(priority::low, [](){ do_something(); }) );
+       */
+      auto& get_priority_queue() { return pri_queue; }
+     
+      bool execute_highest() { return pri_queue.execute_highest(); }
+     
+      void clear() { pri_queue.clear(); }
+
+      /**
+       * Do not run io_service in any other threads, as application assumes single-threaded execution in exec().
+       * @return io_serivice of application
+       */
+      boost::asio::io_service& get_io_service() { return io_serv; }
+
+   private:
+      // members are ordered taking into account that the last one is destructed first
+      boost::asio::io_service  io_serv;
+      execution_priority_queue pri_queue;
+   };
+
+}
+
+using appbase_executor = appbase::default_executor;
```

Comment:
I think this using statement should be in `namespace appbase`

Reply:
Thanks, done.